runtime/src/launch/pbs-aprun/launch-pbs-aprun.c

```diff
@@ -111,18 +111,31 @@ static char* genQsubOptions(char* genFilename, char* projectString, qsubVersion
     walltime = getenv("CHPL_LAUNCHER_WALLTIME");
   }
 
+  // determine the job name
+  char jobName[1024];
+  char* prefix = getenv("CHPL_LAUNCHER_JOB_PREFIX");
+  char* name = getenv("CHPL_LAUNCHER_JOB_NAME");
+  if (prefix == NULL) {
+    prefix = "CHPL-";
+  }
+  if (name == NULL) {
+    snprintf(jobName, sizeof(jobName), "%s%.10s ", prefix, genFilename);
+  } else {
+    strncpy(jobName, name, sizeof(jobName));
+  }
+
```

Comment:
The code that's here looks good, but I'm wondering if we can remove the duplication across launchers by stuffing this into a common routine. Something like what we do for `chpl_get_real_binary_name()`